### ISSUE TITLE
Accessibility issues with RadDataGrid

### ISSUE THREAD

**Issue by mrlacey** [telerik/UI-For-UWP#466]
### Description

RadDataGrid contains multiple accessibility issues.

### Steps to Reproduce

1. Create a UWP app with [Windows Template Studio](https://aka.ms/wtsinstall)
2. Add the TelerikDataGrid page.
3. Run the generated app
4. Analyze with [Accessibility Insights for Windows](https://accessibilityinsights.io/docs/en/windows/overview)

### Expected Behavior

The control should be fully accessible and pass the basic checks, especially the ones that relate to [Section 508](https://www.access-board.gov/guidelines-and-standards/communications-and-it/about-the-ict-refresh/final-rule/single-file-version#502-interoperability-assistive-technology)

Additionally, it should be possible to navigate the content of the grid with Narrator and access all content.
Narrator should also include meaningful names for the elements that can be focused.

### Actual Behavior

Accessibility Insights reports that:
- The control does not support the [Grid pattern](https://docs.microsoft.com/en-us/windows/win32/winauto/uiauto-implementinggrid) for accessible controls.
- The control does not support the [Table pattern](https://docs.microsoft.com/en-us/windows/win32/winauto/uiauto-implementingtable) for accessible controls.
- Onscreen elements to do provide a BoundingRectangle
- Internal to the control, there is an unnamed progressbar that can receive focus.

When using Narrator:
- Only the contents of the first column are read aloud.
- An unhelpful description is provided for the panel where column headers can be dragged for grouping.

It seems the issues dismissed in #457 were never addressed.

### Basic Information

- Version with issue: 1.0.1.9
- Last known good version: n/a
- IDE: VS 16.6.4
- UWP SDK: 18362-19041
- Nuget Packages: 1.0.1.9

### Screenshots

![image](https://user-images.githubusercontent.com/189547/88386888-37f21300-cda9-11ea-827c-761e5898cf91.png)

![image](https://user-images.githubusercontent.com/189547/88386854-26a90680-cda9-11ea-990a-92c95bb55ecb.png)

**Comment by APopatanasov** [comment]
Hi @mrlacey ,

We agree that the accessibility of the DataGrid control could further be improved. #457 was closed because the described issue was tested against an example that uses a very old version of the controls where the DataGrid's accessibility was not even implemented.

I am a little bit confused by the results of the Accessibility Insights tool because the [DataGrid implements both the Grid and Table patters](https://github.com/telerik/UI-For-UWP/blob/12f987f86e2b70e1e87b1d197f4e2d70c0ce34c5/Controls/Grid/Grid.UWP/AutomationPeers/RadDataGridAutomationPeer.cs#L16). The error is about the DataGridContentLayer that should not support it - I believe the error comes from the [GetAutomationControlType core that returns a wrong type for the content layer](https://github.com/telerik/UI-For-UWP/blob/12f987f86e2b70e1e87b1d197f4e2d70c0ce34c5/Controls/Grid/Grid.UWP/AutomationPeers/DataGridContentLayerPanelAutomationPeer.cs#L39).

About the Narrator I have tested the control using it and on selection of a cell the content of the cell was read aloud as expected - there are some issues when Tab navigation is used and there we can improve the control. The accessibility of the grouping could be improved as well.

**Comment by mrlacey** [comment]
When I try and use Narrator, I can't navigate to anything other than the first column.

Using this XAML (if it makes a difference).

```xml
<tg:RadDataGrid ColumnDataOperationsMode="Flyout" x:Name="grid" ItemsSource="{x:Bind Source, Mode=OneWay}" AutoGenerateColumns="False" >
    <tg:RadDataGrid.Columns>
        <tg:DataGridTextColumn PropertyName="OrderID" />
        <tg:DataGridDateColumn PropertyName="OrderDate" />
        <tg:DataGridTextColumn PropertyName="Company" />
        <tg:DataGridTextColumn PropertyName="ShipTo" />
        <tg:DataGridNumericalColumn PropertyName="OrderTotal" />
        <tg:DataGridTextColumn PropertyName="Status" />
        <tg:DataGridTemplateColumn Header="Symbol">
            <tg:DataGridTemplateColumn.CellContentTemplate>
                <DataTemplate x:DataType="model:SampleOrder">
                    <FontIcon
                            HorizontalAlignment="Left"
                            Margin="{StaticResource MediumLeftRightMargin}"
                            FontFamily="{ThemeResource SymbolThemeFontFamily}"
                            Glyph="{x:Bind Symbol}" />
                </DataTemplate>
            </tg:DataGridTemplateColumn.CellContentTemplate>
        </tg:DataGridTemplateColumn>
    </tg:RadDataGrid.Columns>
</tg:RadDataGrid>
```

**Comment by APopatanasov** [comment]
The provided code snippet looks good to me. I have tested the examples that are part of [our SDKExamples application](https://github.com/telerik/UI-For-UWP/tree/development/SDKExamples.UWP) with the narrator and there the Cells are read as expected when they are selected using the mouse - if you are using the Tab for selection/navigation an issue will be observed as explained above (this is the area of the DataGrid that could be improved).
